src/main/java/com/acquia/http/HMACHttpServlet.java

```diff
@@ -219,12 +235,13 @@ private void appendServerResponseValidation(CharRequestWrapper wrappedRequest,
         }
     }
 
-    /** 
+    /**
      * Returns the secret key for the given access key.
      * 
-     * @param accessKey Access Key
-     * @return Secret Key
+     * @param accessKey
+     * @return
+     * @throws Exception
      */
-    protected abstract String getSecretKey(String accessKey);
+    protected abstract String getSecretKey(String accessKey) throws Exception;
```

Comment:
Add a specific exception and include error_code constant (ERROR_NOT_FOUND, ERROR_CANT_RETRIEVE)?